### PR TITLE
feat: download binaries for cred helpers

### PR DESCRIPTION
**Pull request by thedadams**

### CODE REVIEW

File: pkg/repos/get.go [hunk 1]
```diff
@@ -99,27 +97,28 @@ func (m *Manager) EnsureCredentialHelpers(ctx context.Context) error {
...
-		helperName = cliCfg.CredentialsStore
-		suffix     string
+		helperName       = cliCfg.CredentialsStore
+		distInfo, suffix string
 	)
-	if helperName == "wincred" {
-		suffix = ".exe"
-	}
-
 	// The file helper is built-in and does not need to be compiled.
```
Comment (g-linville) [inline pkg/repos/get.go:112]: super nit

```suggestion
	// The file helper is built-in and does not need to be downloaded.
```

File: pkg/repos/get.go [hunk 2]
```diff
@@ -152,37 +145,44 @@ func (m *Manager) deferredSetUpCredentialHelpers(ctx context.Context, cliCfg *co
...
+		if err = os.WriteFile(m.credHelperDirs.RevisionFile, []byte(tool.Source.Repo.Root+tag), 0644); err != nil {
 			return err
 		}
 	} else if err != nil {
 		return err
 	}
 
-	if !needsBuild {
+	if !needsDownloaded {
 		// Check for the existence of the gptscript-credential-osxkeychain binary.
```
Comment (g-linville) [inline pkg/repos/get.go:175]: nit:

```suggestion
		// Check for the existence of the credential helper binary.
```

(I should have fixed this myself before but evidently missed it.)

File: pkg/repos/get.go [hunk 3]
```diff
@@ -152,37 +145,44 @@ func (m *Manager) deferredSetUpCredentialHelpers(ctx context.Context, cliCfg *co
...
 		// If it's there, we have no need to build it and can just return.
-		if _, err := os.Stat(filepath.Join(m.credHelperDirs.BinDir, "gptscript-credential-"+helperName+suffix)); err == nil {
+		if _, err = os.Stat(filepath.Join(m.credHelperDirs.BinDir, "gptscript-credential-"+helperName+suffix)); err == nil {
 			return nil
 		}
 	}
 
 	// Find the Go runtime and use it to build the credential helper.
```
Comment (g-linville) [inline pkg/repos/get.go:182]: nit:

```suggestion
	// Find the Go runtime and use it to download the credential helper.
```

**Review by g-linville — APPROVED** [review]
Just comments changes

**Review by StrongMonkey — APPROVED** [review]

**Review by iwilltry42 — APPROVED** [review]

File: pkg/repos/get.go (hunk 1 above)
Comment (thedadams) [reply pkg/repos/get.go]: Done

File: pkg/repos/get.go (hunk 2 above)
Comment (thedadams) [reply pkg/repos/get.go]: Done

File: pkg/repos/get.go (hunk 3 above)
Comment (thedadams) [reply pkg/repos/get.go:182]: Done

**Pull request merged**